src/runtime_src/core/pcie/driver/linux/xocl/subdev/flash.c

```diff
@@ -903,7 +906,7 @@ static bool is_valid_offset(struct xocl_flash *flash, loff_t off)
 	/* Assuming all flash are of the same size, we use
 	 * offset into flash 0 to perform boundary check. */
 	faddr.slave = 0;
-	return flash_faddr2offset(&faddr) < flash->flash_size;
+	return flash_faddr2offset(&faddr) <= flash->flash_size;
```

Comment:
Could you elaborate more on why the "equal" is needed.   I ask because only flash addresses (e.g., 0 to size of flash) makes sense.  Having it less then or equal "feels" wrong. 

For example: 
If the flash size is 0x10 bytes, what does it mean to have an address of 0x10.

Reply:
you're right. this does not make sense. i'll revert it back.